### PR TITLE
feat: PK sales channels management

### PR DESCRIPTION
**Pull request by fPolic**
**What**
- add support for sales channels management:
  - allow adding sales channels while creating a PK
  - edit sales channels for an existing PK

**Note**
- depends on https://github.com/medusajs/medusa/pull/2810
---

RESOLVES CORE-822

### CODE REVIEW

File: src/domain/publishable-api-keys/modals/add-sales-channels.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        </div>
+        {/* === DIVIDER === */}
+
+        <div className="flex-grow">
+          <div className="flex justify-between items-center gap-2">
+            <InputField
+              name="name"
+              type="string"
+              value={search}
+              className="my-4 h-[40px]"
```
Comment (srindom) [inline src/domain/publishable-api-keys/modals/add-sales-channels.tsx:79]: Figma file notes 32px in button and input height

Comment (srindom) [inline src/domain/publishable-api-keys/modals/add-sales-channels.tsx:79]: suggestion: move `my-4` to container

File: src/domain/publishable-api-keys/modals/add-sales-channels.tsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  }, [props.isVisible])
+
+  const onSave = () => {
+    setSelectedChannels(_selectedChannels)
+    props.close()
+  }
+
+  return (
+    <SideModal close={close} isVisible={!!isVisible}>
+      <div className="flex flex-col justify-between h-[100%] p-6">
```
Comment (srindom) [inline src/domain/publishable-api-keys/modals/add-sales-channels.tsx:60]: suggestion: `h-[100%]` -> `h-full`

**Review by srindom — APPROVED** [review]
Works great - only have a few notes on styling.

The implementation deviates a bit from the design -- it doesn't need to block this PR landing, but let me know if this is on purpose and, if so, how and when we will align the two in future iterations. Also fine with me if you want to move fixes to a separate PR.

I will provide the deviations I found here:

1. Sales Channel Table column for title is too narrow. 
CleanShot 2022-12-19 at 11 08 08

2. Dismiss button should have border
CleanShot 2022-12-19 at 11 08 41

3. Remove and close should not be visible
CleanShot 2022-12-19 at 11 09 12

Note: I don't think the remove functionality has been considered in the design. The "Remove and close" button creates bad UX after you have added sales channels, as it intuitively feels like this is the one to click -- however, it is disabled, and the copy doesn't align with the action that feels correct. My suggestion would be: hide the button when no existing channels are selected so the footer only shows the "Close" button. If a channel is selected, we show the removal option in the table header. [There is a link for it here](https://www.figma.com/file/pOGqKSXwKcS1pIYp8d6ECF/API-key-management-Q4-2022?node-id=5713%3A1879&t=hKUSmu8qTSjzBpJc-11) cc: @ludvig18 :)

**Comment by fPolic** [comment]
> The implementation deviates a bit from the design -- it doesn't need to block this PR landing, but let me know if this is on purpose and, if so, how and when we will align the two in future iterations. Also fine with me if you want to move fixes to a separate PR.

@srindom I will implement the "remove functionality" changes in this PR and then see if @ludvig18 is up for a short review call to polish the details

File: src/domain/publishable-api-keys/tables/sales-channels-table.tsx [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+    accessor: "name",
+    maxWidth: 160,
+    width: 160,
+    Cell: ({ row }) => {
+      return (
+        <span
+          title={row.original.description}
+          className="truncate block max-w-[160px]"
+        >
+          {row.original.description}
```
Comment (srindom) [inline src/domain/publishable-api-keys/tables/sales-channels-table.tsx:49]: todo: should be title

**Review by srindom — APPROVED** [review]
LGTM!

**Review by kasperkristensen — APPROVED** [review]
LGTM great work Frane!

**Pull request merged**